### PR TITLE
Batch DataPlane 2021-06-01.14 update

### PR DESCRIPTION
**Pull request by paterasMSFT**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [x] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by chiragg4u** [comment]
Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.

Please create new branch with this recommendation for the review to continue.

**Review by chiragg4u — DISMISSED** [review]
Please go through the contribution checklist and ARM API Review Checklist added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

**Comment by paterasMSFT** [comment]
> Please go through the contribution checklist and ARM API Review Checklist added as part of the review and mark them completed.

Thank you. I created the PR as a draft as I hadn't done that yet. I will complete the checklist before I mark it ready for review.

> Please create new branch with this recommendation for the review to continue.

I just rebased the branch on top of a commit with the required copy. Is it still necessary for me to create a new branch?

File: .vscode/launch.json [hunk 1]
```diff
@@ -24,4 +24,4 @@
       "program": "${file}"
     }
   ]
-}
\ No newline at end of file
+}
```
Comment (dpwatrous) [inline .vscode/launch.json:27]: Looks like this line snuck into this PR by mistake

File: specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json [hunk 2]
```diff
@@ -14197,14 +14197,14 @@
       "title": "Specifies the ephemeral disk placement for operating system disk for all compute nodes (VMs) in the pool.",
       "description": "This property can be used by user in the request to choose which location the operating system should be in. e.g., cache disk space for Ephemeral OS disk provisioning. For more information on Ephemeral OS disk size requirements, please refer to Ephemeral OS disk size requirements for Windows VMs at https://docs.microsoft.com/en-us/azure/virtual-machines/windows/ephemeral-os-disks#size-requirements and Linux VMs at https://docs.microsoft.com/en-us/azure/virtual-machines/linux/ephemeral-os-disks#size-requirements",
       "enum": [
-        "cachedisk"
+        "cacheDisk"
```
Comment (dpwatrous) [inline specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json:14211]: Unless the REST API is actually going to return "cacheDisk" and not "cachedisk", this should be added as an exception to cspell.

File: specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json [hunk 3]
```diff
@@ -14119,7 +14119,18 @@
           ],
           "x-ms-enum": {
             "name": "StatusLevelTypes",
-            "modelAsString": false
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "error"
```
Comment (dpwatrous) [inline specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json:14125]: There's a case mismatch between these values and the ones in the "enum" property. I *think* we determined that the UpperCamelCase values were correct.

**Review by dpwatrous — CHANGES_REQUESTED** [review]

File: specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json (hunk 2 above)
Comment (paterasMSFT) [reply specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json:14211]: I couldn't find any instances of the all-lowers case in the service code. And actually, they're all 'CacheDisk', including the same enum for the control plane, so I think I'm going to change it to that.

File: specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json (hunk 3 above)
Comment (paterasMSFT) [reply specification/batch/data-plane/Microsoft.Batch/stable/2021-06-01.14.0/BatchService.json:14125]: Thanks. I put it back but forgot to adjust the case.

**Comment by paterasMSFT** [comment]
The breaking change errors pertain to changing the type from nothing ('') to 'object'. I do not believe this is a breaking change.

**Comment by paterasMSFT** [comment]
The breaking change errors pertain to changing the type from nothing ('') to 'object'. I do not believe this is a breaking change.

**Comment by jafreck** [comment]
This cannot be merged until `MaxParallelTasks` has backing FE code to support it.

**Comment by pilor** [comment]
ARM does not review dataplane APIs.  Please reach out to azureapirbcore

**Comment by paterasMSFT** [comment]
@tjprescott could we get a review here? There's some urgency to it. Thank you!

**Comment by paterasMSFT** [comment]
@chiragg4u would you mind approving your requested changes, so I can get a final approval from @tjprescott, who is assigned to the PR?

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@chiragg4u @dpwatrous you both requested changes. Please verify those have been made and approve so we can merge this PR.

**Review by dpwatrous — APPROVED** [review]

**Comment by dpwatrous** [comment]
@chiragg4u looks good to me. Can you approve and merge?

**Review by mikekistler — APPROVED** [review]
These changes look good to me. 👍 

I was originally concerned about the large number of "Cross Version Breaking Changes" reported by the tooling (> 500).  But upon investigation it appears that all these are caused by the addition of `type: object` to a schema that was _already_ an object by virtue of having `properties`, but the tool still reports this as a breaking change.

**Comment by paterasMSFT** [comment]
@tjprescott looks like we're ready to merge.

**Comment by paterasMSFT** [comment]
> These changes look good to me. 👍
> 
> I was originally concerned about the large number of "Cross Version Breaking Changes" reported by the tooling (> 500). But upon investigation it appears that all these are caused by the addition of `type: object` to a schema that was _already_ an object by virtue of having `properties`, but the tool still reports this as a breaking change.

@mikekistler or @JeffreyRichter, mind signing off on the validator "breaking changes" for the [ControlPlane PR](https://github.com/Azure/azure-rest-api-specs/pull/14862) for this release as well, as they're essentially the same thing as on this one? Thanks!

**Pull request merged**